sdk/cosmos/azure-cosmos/azure/cosmos/_routing/routing_range.py

```diff
@@ -195,10 +194,35 @@ def overlaps(range1, range2):
         cmp1 = Range._compare_helper(range1.min, range2.max)
         cmp2 = Range._compare_helper(range2.min, range1.max)
 
-        if cmp1 <= 0 or cmp2 <= 0:
+        if cmp1 <= 0 and cmp2 <= 0:
```

Comment:
It was essentially saying more ranges overlap then what actually do. In other words, the previous logic would have false positives. Take the following two ranges Range("3F", "7F", True, False) and Range("", "2F", True, False). These don't overlap, but with or it would pass. Added tests for scenarios like these. The `and` check here is essentially the check for overlapping. The `or` check below is checking for the bounds, so when one of the boundaries are equal it looks if it is minExclusive and maxExclusive sort of thing.

if overlaps:
    if share a boundary and boundaries don't match exclusive and inclusive:
            return false
    return true 
return false